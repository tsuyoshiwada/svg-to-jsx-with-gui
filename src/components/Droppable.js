import React from 'react';
import styled from 'styled-components';
import withDroppable from './hoc/withDroppable';


const DropzoneOverlay = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 9999;
  background: rgba(255, 255, 255, 0.96);

  &::before {
    position: absolute;
    top: 20px;
    right: 20px;
    bottom: 20px;
    left: 20px;
    z-index: 1;
    border: 2px dashed rgba(0, 0, 0, 0.2);
    border-radius: 10px;
    content: '';
  }

  & > div {
    position: relative;
    z-index: 2;
    text-align: center;
  }

  & h2 {
    margin: 0 0 0.2em;
    font-size: 2.87rem;
  }

  & p {
    margin: 0;
    font-size: 0.87rem;
  }
`;

const Droppable = ({ children, isDragOver }) => (
  <div>
    {isDragOver &&
      <DropzoneOverlay>
        <div>
          <h2>DRAG &amp; DROP</h2>
          <p>Let&#39;s drop your svg file!</p>
        </div>
      </DropzoneOverlay>
    }

    {children}
  </div>
);

export default withDroppable(Droppable);
