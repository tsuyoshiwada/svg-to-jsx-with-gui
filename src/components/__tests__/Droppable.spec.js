import React from 'react';
import { shallow } from 'enzyme';
import { Droppable } from '../Droppable';


describe('components#<Droppable />', () => {
  test('Should be render', () => {
    const wrapper = shallow(<Droppable isDragOver={false}>Foo</Droppable>);

    expect(wrapper).toMatchSnapshot();
  });


  test('Should be render overlay', () => {
    const wrapper = shallow(<Droppable isDragOver>Bar</Droppable>);

    expect(wrapper).toMatchSnapshot();
  });
});
