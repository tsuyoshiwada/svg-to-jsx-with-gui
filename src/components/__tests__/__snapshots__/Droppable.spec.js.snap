// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components#<Droppable /> Should be render 1`] = `
<div>
  Foo
</div>
`;

exports[`components#<Droppable /> Should be render overlay 1`] = `
<div>
  <styled.div>
    <div>
      <h2>
        DRAG & DROP
      </h2>
      <p>
        Let's drop your svg file!
      </p>
    </div>
  </styled.div>
  Bar
</div>
`;
